### PR TITLE
Spamming throwing items fix

### LINKED ISSUE #22741: Thrown Items have no attack cooldown
#### Description of issue

[exploitable]
See title, with a throw hotkey (and some spicy third party software), you can easily throw half a dozen shurikens in a second, which is relatively strong all things considered

### PR DESCRIPTION
**Pull request by DDihanov**
[exploitable][tested][tweak]

Hey, it works ¯\\_(ツ)_/¯
closes #22741

:cl:
 * tweak: Added small delay to item throwing.

### CODE REVIEW

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
1.5 seconds seems too much

**Comment by DDihanov** [comment]
1 second is literally nothing - the amount of time you need to click on the other item, press R and click to throw it amounts for 1 second, literally adding 0 delay. 2 is too much. But 1.5 is enough to be noticeable. Pull the branch and try it out. I can lower the delay. But you need to see it for yourself.

**Comment by brndd** [comment]
I think 1 second is more than enough considering the intention of the change should be to stop autoclickers from being absolutely ridicilous with certain items (read: shurikens).

**Comment by DDihanov** [comment]
Ah autoclickers. People actually use that? Alright I'll set it to 1.

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -1634,6 +1634,9 @@ Thanks.
 #define THREW_NOTHING -1
 
 /mob/living/throw_item(var/atom/target,var/atom/movable/what=null)
+	if (src.throw_delayer.blocked())
+		return
```
Comment (PaulRitter) [inline code/modules/mob/living/living.dm]: why not `return FAILED_THROW`?

**Review by PaulRitter — CHANGES_REQUESTED** [review]

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I have definitely thrown stuff faster than 1 second in non-combat applications, like when collecting stuff into a locker
It seems to me like a bad idea to add a delay so large that it will inconvenience regular players over something that only became a problem as an edge case (the ninja shurikens which are thrown differently from any other stacked item)

**Comment by brndd** [comment]
If 1 second is too slow, the delay can be reduced to half a second. If even that is too slow, it can be reduced to 0.25 seconds. The point is to have *some* delay to stop macro abuse, and regardless of how fast you are you will never be as fast as a macro.

**Comment by Kurfursten** [comment]
I agree with brndd and bauds, there should be a delay, but let's make it a delay that focuses on limiting autoclickers not on hurting QoL for legitimate players.

**Comment by DDihanov** [comment]
So 0.25?

**Comment by brndd** [comment]
I would go with 0.3 because Byond counts in deciseconds so you probably can't get half-decisecond accuracy.

EDIT: or alternatively lower than that at 0.2. Three or four throws a second is *probably* about the limit of what a human can reasonably do, though.

**Comment by brndd** [comment]
You'll want 3, not 0.3. Deciseconds, because Byond.

**Comment by Kurfursten** [comment]
0.3 is fine by me, even though I consistently click around 7 times per second at my maximum speed which is a little over twice that, but I think it should be okay. We can keep an eye on it to see how it feels and adjust it if necessary.

https://www.click-test.com/p/game.html

**Comment by brndd** [comment]
But how fast do you press R and then click?

**Comment by Exxion** [comment]
If you have two hands it should be pretty much the same

**Comment by brndd** [comment]
I think you'll find it's difficult to make sure you do this in exact sequence. You can't really just randomly mash them: if you press R twice and only click once, you won't throw.

**Comment by Exxion** [comment]
Yeah but you can usually keep it close enough to get at least a few in a row unless you're going at absolute maximum speed
You might have to slow down a bit but it's still going to be faster than 3 per second
But also shuriken are the only thing you can throw repeatedly like that at all

I have absolutely no idea where I'm going with this post

**Comment by Blithering** [comment]
Whats the standard attack delay? Could go off that for the timer

**Pull request merged**